core/test/Test/Pos/Core/Bi.hs

```diff
@@ -0,0 +1,1739 @@
+    attrs = AddrAttributes hap BootstrapEraDistr
+    hap = Just (HDAddressPayload (getBytes 32 32))
+
+roundTripAddressBi :: Property
+roundTripAddressBi = eachOf 1000 genAddress roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- AddrSpendingData
+--------------------------------------------------------------------------------
+
+golden_AddrSpendingData_PubKey :: Property
+golden_AddrSpendingData_PubKey = goldenTestBi exampleAddrSpendingData_PubKey
+                                              "test/golden/AddrSpendingData_PubKey"
+
+golden_AddrSpendingData_Script :: Property
+golden_AddrSpendingData_Script = goldenTestBi asd "test/golden/AddrSpendingData_Script"
+  where asd = ScriptASD exampleScript
+
+golden_AddrSpendingData_Redeem :: Property
+golden_AddrSpendingData_Redeem = goldenTestBi asd "test/golden/AddrSpendingData_Redeem"
+  where
+    asd = RedeemASD redeemPublicKey
+    Just redeemPublicKey = fst <$> redeemDeterministicKeyGen (getBytes 0 32)
+
+golden_AddrSpendingData_Unknown :: Property
+golden_AddrSpendingData_Unknown = goldenTestBi asd "test/golden/AddrSpendingData_Unknown"
+  where asd = UnknownASD 247 (getBytes 3 32)
+
+roundTripAddrSpendingDataBi :: Property
+roundTripAddrSpendingDataBi = eachOf 1000 genAddrSpendingData roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- AddrStakeDistribution
+--------------------------------------------------------------------------------
+
+golden_AddrStakeDistribution_Bootstrap :: Property
+golden_AddrStakeDistribution_Bootstrap =
+    goldenTestBi BootstrapEraDistr "test/golden/AddrStakeDistribution_Bootstrap"
+
+golden_AddrStakeDistribution_SingleKey :: Property
+golden_AddrStakeDistribution_SingleKey =
+    goldenTestBi asd "test/golden/AddrStakeDistribution_SingleKey"
+  where
+    asd = SingleKeyDistr (abstractHash examplePublicKey)
+
+golden_AddrStakeDistribution_UnsafeMultiKey :: Property
+golden_AddrStakeDistribution_UnsafeMultiKey =
+    goldenTestBi asd "test/golden/AddrStakeDistribution_UnsafeMultiKey"
+  where
+    asd   =  M.fromList (zip sis coins) :: Map StakeholderId CoinPortion
+    sis   = [si1, si2, si3]
+    coins = map (CoinPortion . exp10_14) [3,2,5]
+    exp10_14 x = x * (10 :: Word64) ^ (14 :: Word64)
+    Right si1 = abstractHash . PublicKey <$> xpub (getBytes  0 64)
+    Right si2 = abstractHash . PublicKey <$> xpub (getBytes 13 64)
+    Right si3 = abstractHash . PublicKey <$> xpub (getBytes 27 64)
+
+roundTripAddrStakeDistributionBi :: Property
+roundTripAddrStakeDistributionBi = eachOf 1000 genAddrStakeDistribution roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- AddrType
+--------------------------------------------------------------------------------
+
+golden_AddrType_PK :: Property
+golden_AddrType_PK = goldenTestBi at "test/golden/AddrType_PK"
+  where at = ATPubKey
```

Comment:
Why not just replace `at` with `ATPubKey`? Saves you 7 characters :)

Reply:
Yeah good call, I must have gotten into copy-paste mode 😅